pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -122,6 +123,8 @@ type Interface interface {
 	// Note that both `state` and `nodeInfo` are deep copied.
 	SelectVictimsOnNode(ctx context.Context, state *framework.CycleState,
 		pod *v1.Pod, nodeInfo *framework.NodeInfo, pdbs []*policy.PodDisruptionBudget) ([]*v1.Pod, int, *framework.Status)
+	// DeleteVictim deletes the given victim pod from API server for preemption.
+	DeleteVictim(ctx context.Context, cs kubernetes.Interface, victim, preemptor *v1.Pod) error
```

Comment:
preemptor is not used at all, can you remove it from the function signature?

Reply:
True that its not used by the default plugin, but it is useful for cases beyond the trivial default case. In my use case, I annotate the victim pod info with the name of the preemptor so I can tell what pod caused it to get deleted.

In terms of API design, I would consider the preemptor to be a part of the context for the victim deletion.

Reply:
> In my use case, I annotate the victim pod info with the name of the preemptor so I can tell what pod caused it to get deleted.

Just curious how do you annotate this info? and if you're exposing the preemptor's info to the victim, that means the end-user of the victim is able to view the preemptor's info, which is sort of breaking the security boundary as #114882 mentioned.

Reply:
We are a single namespace system so don't have that particular bug, though as the bug calls out one could check that the namespaces are the same before annotating data. 

In our case we would like to annotate the pod with the preemptor info. I see why you removed it from the shared preemption library, this change allows a plugin implementation to do it on our end.

Reply:
I'm still preferred to not pass preemptor in DeleteVictim(), to find out the preemption info, L380 in pkg/scheduler/framework/preemption/preemption.go can give you enough info.

Reply:
Is this a hard blocker? On my end, we prefer pod annotations to events and would like the capability to annotate the pod with the preemptor rather than subscribe to the event.

Reply:
+1 to remove the preemptor since it's not used in Kubernetes code base and your case is somehow break the security boundary. Sorry, but it's actually the only case I heard from the community, maybe they do exist, I have no idea.

> this change allows a plugin implementation to do it on our end

Besides, as I mentioned above, if there's no other cases, I'm not convinced of adding a new interface method, considering this will break current implementations but no behavior changes. cc @Huang-Wei 

Let's push this forward.